packages/nextjs/src/tools/codegen/utils.ts

```diff
@@ -47,7 +46,7 @@ export const validateDeployContext = () => {
   if (process.env.VERCEL && !process.env.VERCEL_REGION) {
     return true;
   }
-  if (process.env.SITECORE && process.env.BuildMetadata_BuildId) {
+  if (process.env.SITECORE && process.env.SITECORE_BUILD) {
```

Comment:
The request was sent for Deploy to add an environment variable, reliably indicating deploy/build context, with this name